### PR TITLE
newCAFromRaw(): create ComboAddress from raw 4/16 byte strings

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
~still needs docs~

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by Habbie** [comment]
dnsdist-lua-bindings.cc already contains a variant of this, which I had not noticed.

**Comment by Habbie** [comment]
> dnsdist-lua-bindings.cc already contains a variant of this, which I had not noticed.

Copied that one now.

File: regression-tests.auth-py/test_LuaRecords.py [hunk 1]
```diff
@@ -598,6 +599,23 @@ def testANY(self):
...
+        """
+        Test newCAFromRaw() function
+        """
+        name = 'newcafromraw.example.org.'
+
+        query = dns.message.make_query(name, 'A')
+
+        response = dns.message.make_response(query)
+
+        response.answer.append(dns.rrset.from_text(name, 0, dns.rdataclass.IN, dns.rdatatype.A, '65.66.67.68'))
```
Comment (omoerbeek) [inline regression-tests.auth-py/test_LuaRecords.py:612]: Please also add an AAAA test

Comment (Habbie) [reply regression-tests.auth-py/test_LuaRecords.py:613]: done!

File: docs/lua-records/reference/comboaddress.rst [hunk 2]
```diff
@@ -19,6 +19,13 @@ Functions and methods of a ``ComboAddress``
 
     addr = newCA("1.2.3.4")
 
+.. function:: newCAFromRaw(rawaddress[, port]) -> ComboAddress
+
+  Returns a new :class:`ComboAddress` object based on the 4- or 16-octet string.
+
+  :param string rawaddress: The IPv4 of IPv6 address as a 4/16 octet string
```
Comment (omoerbeek) [inline docs/lua-records/reference/comboaddress.rst:26]: Example or more explanation needed, it's not clear how to encode the string

Comment (Habbie) [reply docs/lua-records/reference/comboaddress.rst:27]: added an example

**Review by omoerbeek — APPROVED** [review]

**Comment by Habbie** [comment]
rebased

**Comment by Habbie** [comment]
The circleci commit is also in #10459

**Pull request merged**